src/light_client.rs

```diff
@@ -283,84 +273,45 @@ pub async fn process_block(
 					.collect::<Vec<_>>();
 				rpc_fetched.extend(partition_fetched_filtered.clone());
 			}
+			let rpc_call_duration = begin.elapsed();
+			let rpc_fetched_len = rpc_fetched.len();
+			info!(
+				block_number,
+				"partition_rpc_retrieve_time_elapsed" = ?rpc_call_duration,
+				"partition_rpc_cells_fetched" = rpc_fetched_len,
+				"Partition cells received from RPC",
+			);
+			metrics
+				.record(MetricValue::RPCCallDuration(
+					rpc_call_duration.as_secs_f64(),
+				))
+				.await?;
 		}
 
-		let begin = Instant::now();
-
 		let rpc_fetched_data_cells = rpc_fetched
 			.iter()
 			.filter(|cell| !cell.position.is_extended())
 			.collect::<Vec<_>>();
 		let rpc_fetched_data_rows = data::rows(dimensions, &rpc_fetched_data_cells);
-		let rows_len = rpc_fetched_data_rows.len();
 
-		let dht_insert_rows_success_rate = light_client
+		light_client
 			.insert_rows_into_dht(block_number, rpc_fetched_data_rows)
-			.await;
-		let success_rate: f64 = dht_insert_rows_success_rate.into();
-		let time_elapsed = begin.elapsed();
-
-		info!(
-			block_number,
-			"DHT PUT rows operation success rate: {dht_insert_rows_success_rate}"
-		);
-
-		metrics
-			.record(MetricValue::DHTPutRowsSuccess(success_rate))
-			.await?;
-
-		info!(
-			block_number,
-			"partition_dht_rows_insert_time_elapsed" = ?time_elapsed,
-			"{rows_len} rows inserted into DHT"
-		);
-
-		metrics
-			.record(MetricValue::DHTPutRowsDuration(time_elapsed.as_secs_f64()))
-			.await?;
+			.await
+			.map_err(|e| {
+				warn!("Error inserting rows into DHT: {e}");
+				e
+			})
+			.ok();
```

Comment:
Since we are not returning error, we can use `if let Err(error) = ...` here, to avoid confusion. But previous code was returning error, so is this intentional?

Reply:
Yeah, we wan't to log the errors but we don't want to stop the block processing loop because of PUT errors. At least for now...